src/components/multi-action-button/multi-action-button.test.tsx

```diff
@@ -0,0 +1,157 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import MultiActionButton from "./multi-action-button.component";
+import Button from "../button";
+import { testStyledSystemMargin } from "../../__spec_helper__/test-utils";
+
+test("should render with provided 'text'", () => {
+  render(
+    <MultiActionButton text="Main Button">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByRole("button", { name: "Main Button" })).toBeVisible();
+});
+
+test("should render with provided 'subtext' when 'size' is 'large'", () => {
+  render(
+    <MultiActionButton text="Main Button" subtext="Subtext" size="large">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByText("Subtext")).toBeVisible();
+});
+
+test("should render with provided 'data-' tags", () => {
+  render(
+    <MultiActionButton text="Main Button" data-element="foo" data-role="bar">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByTestId("bar")).toHaveAttribute("data-element", "foo");
+});
+
+test("should render with provided 'id'", () => {
+  render(
+    <MultiActionButton text="Main Button" id="test">
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  expect(screen.getByRole("button", { name: "Main Button" })).toHaveAttribute(
+    "id",
+    "test"
+  );
+});
+
+test("should call 'onClick' when the main button is clicked", async () => {
+  const onClick = jest.fn();
+  const user = userEvent.setup();
+  render(
+    <MultiActionButton text="Main Button" onClick={onClick}>
+      <Button>First</Button>
+    </MultiActionButton>
+  );
+
+  await user.click(screen.getByRole("button", { name: "Main Button" }));
```

Comment:
**suggestion** (non-blocking): could define a `mainButton` const instead of locating the main button within the `.click` to increase readability here

Reply:
You can apply this comment to anywhere we're finding a main button, first button etc.

Reply:
imo I don't think thats really necessary in these cases since its only being used once and the test description also states what button is being clicked, I do think its worth doing if the same element is being used several times.

Reply:
Yeah, just personal preference, happy for you to leave as is